tests/processor/test_tasks.py

```diff
@@ -146,16 +157,125 @@ def test_get_report_files_second_task_not_called(self,
+        mock_process_files.delay.assert_not_called()
+
+    @patch('masu.processor.tasks.ReportStatsDBAccessor.get_last_completed_datetime')
+    @patch('masu.processor.tasks.ReportStatsDBAccessor.get_last_started_datetime')
+    @patch('masu.processor.tasks._get_report_files')
+    @patch('masu.processor.tasks.process_report_file')
+    def test_get_report_files_timestamps_aligned(self,
+                                                 mock_process_files,
+                                                 mock_get_files,
+                                                 mock_started,
+                                                 mock_completed):
+        """
+        Test that the chained task is called when start timestamp is before
+        end timestamp.
+        """
+        mock_process_files.delay = Mock()
+        mock_get_files.return_value = self.fake_reports
 
-        account = fake_arn(service='iam', generate_account_id=True)
-        schema_name = self.fake.word()
-        get_report_files(customer_name=self.fake.word(),
-                         authentication=account,
-                         provider_type='AWS',
-                         report_name=self.fake.word(),
-                         schema_name=schema_name,
-                         billing_source=self.fake.word(),
-                         provider_id=random.randint(1,65535))
+        mock_started.return_value = self.yesterday
+        mock_completed.return_value = self.today
+
+        get_report_files(**self.fake_get_report_args)
+        mock_process_files.delay.assert_called()
+
+    @patch('masu.processor.tasks.ReportStatsDBAccessor.get_last_completed_datetime')
+    @patch('masu.processor.tasks.ReportStatsDBAccessor.get_last_started_datetime')
+    @patch('masu.processor.tasks._get_report_files')
+    @patch('masu.processor.tasks.process_report_file')
+    def test_get_report_files_timestamps_misaligned(self,
+                                                    mock_process_files,
+                                                    mock_get_files,
+                                                    mock_started,
+                                                    mock_completed):
+        """
+        Test that the chained task is called when start timestamp is before
+        end timestamp.
+        """
+        mock_process_files.delay = Mock()
+        mock_get_files.return_value = self.fake_reports
+
+        mock_started.return_value = self.today
+        mock_completed.return_value = self.yesterday
+
+        get_report_files(**self.fake_get_report_args)
+        mock_process_files.delay.assert_not_called()
+
+    @patch('masu.processor.tasks.ReportStatsDBAccessor.get_last_completed_datetime')
+    @patch('masu.processor.tasks.ReportStatsDBAccessor.get_last_started_datetime')
+    @patch('masu.processor.tasks._get_report_files')
+    @patch('masu.processor.tasks.process_report_file')
+    def test_get_report_files_timestamps_empty(self,
```

Comment:
There are now 4 tests with the same function name. Only 1 of them will register and run.